src/utils/rollout_utils.ts

```diff
@@ -0,0 +1,30 @@
+import { createHash } from 'crypto';
+
+/**
+ * A function that allows for gradual, consistent rollouts of features
+ *
+ * Given a message, and a target rollout threshold  (between 0 and 1), returns a boolean of whether
+ * the message should get the rolled out treatment or not.
+ *
+ * This method has the property that the same message will always recieve the same treatment, even as
+ * the threshold is increased over time.
+ */
+export const isRolledOut = ({ message, threshold }: { message: string; threshold: number }): boolean => {
```

Comment:
my instinct says this should just be a function expression but i dont have a good reason for it